components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/ChallengeQuestionManager.java

```diff
@@ -1080,6 +1080,7 @@ private void checkChallengeQuestionExists(UserChallengeAnswer[] userChallengeAns
                 Utils.handleClientException(ERROR_CODE_CHALLENG_ANSWER_MISSING, null);
             }
 
+            assert challengeQuestion != null;
```

Comment:
If we throw an exception in the above `challengeQuestion == null` condition, no need of this assert here. Let's modify the above condition.